src/re.rs

```diff
@@ -980,6 +980,21 @@ impl<'t> Index<&'t str> for Captures<'t> {
 
 }
 
+impl<'t> fmt::Debug for Captures<'t> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        try!(writeln!(f, "Captures of {}:", self.text));
+        for i in 0..self.locs.len() {
+            try!(writeln!(f, "{}:\t{:?}", i, self.at(i)));
+        }
+        if let Some(ref h) = self.named {
+            for (n, i) in h.iter() {
+                try!(writeln!(f, "{}:\t{:?}", n, self.at(*i)));
+            }
+        }
+        Ok(())
+    }
+}
```

Comment:
I think the convention is to show the raw type as much as possible. In this case, I suspect it's safe to drop the `text` field (since that could be quite large and distracting) but show the `locs` and `named` fields as-is.

I urge you to use the debug formatting helpers: http://doc.rust-lang.org/std/fmt/struct.DebugStruct.html :-)